R/rd-r6.R

```diff
@@ -249,38 +250,38 @@ r6_inherited_method_list <- function(block, r6data) {
     "><summary>Inherited methods</summary>"
   )
 
-  c("\\if{html}{",
-    paste0("\\out{", details, "}"),
-    "\\itemize{",
+  c("\\if{html}{\\out{",
+    "<ul>",
     sprintf(
       paste0(
-        "\\item \\out{<span class=\"pkg-link\" data-pkg=\"%s\" ",
-        "data-topic=\"%s\" data-id=\"%s\">}",
-        "\\href{../../%s/html/%s.html#method-%s}{\\code{%s::%s$%s()}}",
-        "\\out{</span>}"
+        "<li>",
+        "<span class=\"pkg-link\" data-pkg=\"%s\" data-topic=\"%s\" data-id=\"%s\">",
+        "<a href='../../%s/html/%s.html#method-%s-%s'><code>%s::%s$%s()</code></a>",
+        "</li>"
       ),
       super_meth$package,
       super_meth$classname,
       super_meth$name,
       super_meth$package,
       super_meth$classname,
+      super_meth$classname,
       super_meth$name,
       super_meth$package,
       super_meth$classname,
       super_meth$name
     ),
-    "}",
-    "\\out{</details>}",
-    "}"
+    "</ul>",
+    "</details>",
+    "}}"
   )
 }
 
 r6_method_begin <- function(block, method) {
   nm <- r6_show_name(method$name)
   c(
     "\\if{html}{\\out{<hr>}}",
-    paste0("\\if{html}{\\out{<a id=\"method-", nm, "\"></a>}}"),
-    paste0("\\if{latex}{\\out{\\hypertarget{method-", nm, "}{}}}"),
+    paste0("\\if{html}{\\out{<a id=\"method-", method$class, "-", nm, "\"></a>}}"),
```

Comment:
To avoid breaking cross package links, can we define both anchors here?

Reply:
Oh hmmmm, we can't do that because it'll lead to the same HTML 5 error.

Reply:
Oh, right, of course.